imports/ui/unit-explorer/unit-explorer.jsx

```diff
@@ -39,55 +42,84 @@ class UnitExplorer extends Component {
     dispatch(push(`/unit/${id}`))
   }
 
+  findUnit = (searchText) => {
+    if (searchText === '') {
+      this.setState({searchOn: false})
+    } else {
+      this.setState({searchOn: true})
+      const matcher = new RegExp(searchText, 'i')
+      const searchResult = this.props.unitList
+        .filter(unit => !matcher || (unit.name && unit.name.match(matcher)))
+      this.setState({
+        searchResult: searchResult
+      })
+    }
+  }
+
+  searchOff = () => {
+    this.setState({searchOn: false})
+  }
+
   render () {
     const { isLoading, unitList, dispatch, currentUserId } = this.props
-
+    const { searchResult, searchOn } = this.state
     if (isLoading) return <Preloader />
     const activeUnits = unitList.filter(unitItem => unitItem.metaData && !unitItem.metaData.disabled)
     const disabledUnits = unitList.filter(unitItem => unitItem.metaData && unitItem.metaData.disabled)
-
     return (
       <div className='flex flex-column flex-grow full-height'>
-        <RootAppBar title='My Units' onIconClick={() => dispatch(setDrawerState(true))} shadowless />
+        <RootAppBar
+          title='My Units'
+          onIconClick={() => dispatch(setDrawerState(true))}
+          shadowless
+          findItem={this.findUnit}
+          searchOff={this.searchOff}
```

Comment:
Both `searchOn` and `searchOff` are actually redundant in this case.
You can use the text sent to the `findUnit` callback to determine whether you're currently in search mode-
* If it's empty you're not in search mode
* If it's not empty, you are in search mode

This will also make it so that if the user was searching, but then deleted the query, the list tab interface with all the units will reappear (I think that would be a desired behavior)